index.js

```diff
@@ -131,12 +131,35 @@ function handle_social_coding_welcome(event, state, member, client){
 }
 
 function handleDish(event, room, client, auth){
-  const sender = event.getSender();
-  const message = event.getContent().body;
-  
+  let message = event.getContent().body;
+  let matched = false;
+  if (event.getSender() == `@${process.env.BOT_USER}:matrix.org`){ // we sent the message.
+    return;
+  }
+  if (message.trim()[0] == ">"){ //quoting another user, skip the quoted part
+    matched = true;
+    message = message.split("\n\n")[1];
+  }
+
+  const re = /!\s*dish\s+(\S+)\s+(\S+)\s+points\s+to\s+(\S+)\s+for\s+([^\n]+)/g;
+  let match;
+  do {
+    match = re.exec(message.toLowerCase());
```

Comment:
You will only get lower cased parameters from match. So e.g. matching the user would fail because it's case sensitive I think

Reply:
Great point. Fixed 🎉